File: tournaments/factorizedGCD/factorizedGCD.js
function factorizedGCD(a, b) {
    var j = 0,
        result = 1;
    for (var i = 0; i < a.length; i++) {
        while (j < b.length && a[i] > b[j]) {
            j++;
        }
        if (j < b.length && a[i] === b[j]) {
            result *= a[i];
            j++;
        }
    }
    return result;
}
  

File: tournaments/factorizedGCD/factorizedGCD.cpp
int factorizedGCD(std::vector<int> a, std::vector<int> b) {
  int j = 0,
      result = 1;
  for (int i = 0; i < a.size(); i++) {
    while (j < b.size() && a[i] > b[j]) {
      j++;
    }
    if (j < b.size() && a[i] == b[j]) {
      result *= a[i];
      j++;
    }
  }
  return result;
}
